### PR TITLE
Added Imgur support (not tested)

### PR DESCRIPTION
**Pull request by fufik**
Modified README
Add it to dev

### CODE REVIEW

**Comment by 26000** [comment]
Sorry, will need quite some time to integrate it, so not today ^^'
I'll just comment on the code now and will fix it later.

File: config/config.go [hunk 1]
```diff
@@ -13,28 +13,32 @@ const (
 )
 
 // ReadConfig reads the configuration file.
-func ReadConfig(path string) (error, *Irc, *Telegram, *Irchuu) {
+func ReadConfig(path string) (error, *Irc, *Telegram, *Irchuu,*Imgur) {
```
Comment (26000) [inline config/config.go:16]: (just a note to self) Better to add Imgur's config to the Irchuu part.

File: config/config.go [hunk 2]
```diff
@@ -176,4 +182,7 @@ type Telegram struct {
 	DataDir       string
 }
 
+type Imgur struct{
```
Comment (26000) [inline config/config.go:185]: Same

File: imgur/imgur.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"github.com/26000/irchuu/config"
+	"net/http"
+	"gopkg.in/telegram-bot-api.v4"
+	"bytes"
+	"encoding/json"
+)
+
+func upload(bot *tgbotapi.BotAPI, id string, c *config.Imgur) (url string, err error) {
```
Comment (26000) [inline imgur/imgur.go:11]: Funcs with names starting with lowercase letters are private in Golang.

File: imgur/imgur.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package imgur
+
+import (
+	"github.com/26000/irchuu/config"
+	"net/http"
+	"gopkg.in/telegram-bot-api.v4"
+	"bytes"
+	"encoding/json"
+)
+
```
Comment (26000) [inline imgur/imgur.go:10]: Missing docs.

File: imgur/imgur.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+func upload(bot *tgbotapi.BotAPI, id string, c *config.Imgur) (url string, err error) {
+	file, err := bot.GetFileDirectURL(id)
+	if err != nil {
+		return
+	}
+	//Succesfully copied from StackOverFlow by fufik :^)
+	baseurl := "https://api.imgur.com/3/upload"
+	jsonStr := []byte(`{"image":"`+ file +`"}`)
+	req, err := http.NewRequest("POST", baseurl, bytes.NewBuffer(jsonStr))
```
Comment (26000) [inline imgur/imgur.go:19]: Seems nice, needs to be tested.

File: config/config.go [hunk 6]
```diff
@@ -112,6 +116,8 @@ NamesUpdateInterval = 600 # (seconds) how often to poll the server for the
 
 maxhist = 40 # maximum number of messages sent on ./hist command in IRC
              # works only when dbURI is set
+[imgur]
```
Comment (26000) [inline config/config.go:119]: Not needed if in the clientid is in Irchuu.

**Review by 26000 — COMMENTED** [review]
Needs to be integrated yet.

**Comment by 26000** [comment]
Sorry, seems like it doesn't work... Fixable (several mistakes like trying to unmarshal body which is `io.ReadCloser` or trying to get a value from an unindexable type `interface {}`), though it's easier to rewrite from scratch. It isn't a priority to work specifically with imgur, so i'll probably try to integrate with pomfs. I hope i didn't hurt you with taking so much time and rejecting it >_> sorry.

**Comment by fufik** [comment]
Why we are still here? Just to suffer? Every night...I can feel my commit....my code....even functions. The commit I've lost... the structures I've lost.. won't stop hurting.

**Comment by 26000** [comment]
@fufik , eeehh... Sorry, I don't even know what to answer.
Why are we still here? I don't know. If I could, I'd already be somewhere else. Not in this fucking world.

Yeah, just to suffer. Every day is fucking hell you can't even imagine. I can't do anything. IRChuu~ needs new code, new features and bugfixes. I simply fucking can't. I need to relax, but no matter what do I do, I get even more tired. There's nothing that I want to do or that brings me joy.

Mental illnesses are crap which shouldn't exist. It's just dishonest, I dunno how to express that. Tire, helplessness, hopelessness, defenselessness. I wouldn't wish that to my worst enemy... Eeh, my worst enemy already suffers from that — I am the enemy.

I don't see the end, how much I need to wait, how much more do I have to suffer? Why the fuck I even have to? What did I do so wrong? Somebody, kill me please... Every fucking day I dream about that. About the end. Even though I'm trying to be cured for more than a month. And it's not even a bit better.

I feel like I am already dead emotionally. I have some memories, I try to be myself like I was back then. And fail much. I don't thing that could be cured, no pills could help. I wish everybody left me alone and let me die. No way...

Sorry for all this, I couldn't resist. I shouldn't allow myself like that...

**Pull request closed without merging**